rewriter/Util.cc

```diff
@@ -198,7 +198,8 @@ ast::TreePtr ASTUtil::thunkBody(core::MutableContext ctx, ast::TreePtr &node) {
     if (send->fun != core::Names::lambda() && send->fun != core::Names::proc()) {
         return nullptr;
     }
-    if (!send->recv->isSelfReference()) {
+    auto *constRecv = ast::cast_tree<ast::ConstantLit>(send->recv);
+    if (!send->recv->isSelfReference() && constRecv && constRecv->symbol != core::Symbols::Kernel()) {
```

Comment:
```suggestion
    if (constRecv == nullptr || constRecv->symbol != core::Symbols::Kernel()) {
```

Reply:
how about we split it out into two separate conditions?

1. we check if we have a constant receiver which is not `Kernel`, which means this is not a thunk body, so return `nullptr`
2. we check if we have a receiver that is not `self`, which again means this is not a thunk body, so we return `nullptr` (this is the current behaviour)

```suggestion
    auto *constRecv = ast::cast_tree<ast::ConstantLit>(send->recv);
    if (constRecv != nullptr && constRecv->symbol != core::Symbols::Kernel()) {
        return nullptr;
    }
    if (!send->recv->isSelfReference()) {
```

This looks more correct. WDYT?

Reply:
I like this change, thanks!

Reply:
I pushed a new version of this conditional. There's now a utility function called `isKernel` that is used to test the receiver. The conditional now reads:
```cpp
if (!send->recv->isSelfReference() && !isKernel(send->recv)) {
  return nullptr;
}
```